gen_allocdefs.h

```diff
@@ -14,7 +14,11 @@ static int ta_rplus ## _internal (ta *x, unsigned int n, unsigned int pluslen) \
       return 1; } \
     return 1; } \
   x->len = 0; \
-  return !!(x->field = (type *) alloc((x->a = n) * sizeof(type))); } \
+  x->field = alloc(n * sizeof(type)); \
```

Comment:
The conversion is correct, ideally this could would also detect an overflow of this multiplication:

```
if (n && sizeof(type) > (size_t)-1/n) return 0;
```
```

Reply:
I don't want to change the behavior of these functions yet. This can be done in another PR.